makefile

```diff
@@ -87,11 +87,11 @@ beanstalkcpp.o: beanstalk.cc beanstalk.hpp makefile
 
 install: $(SHAREDLIB) $(STATICLIB)
 	install -d $(DESTDIR)$(INCLUDEDIR)
-	install beanstalk.h $(DESTDIR)$(INCLUDEDIR)
-	install beanstalk.hpp $(DESTDIR)$(INCLUDEDIR)
+	install -m 0644 beanstalk.h $(DESTDIR)$(INCLUDEDIR)
+	install -m 0644 beanstalk.hpp $(DESTDIR)$(INCLUDEDIR)
 
 	install -d $(DESTDIR)$(LIBDIR)
-	install -m 0644 $(SHAREDLIB) $(DESTDIR)$(LIBDIR)/$(SHAREDLIB).$(VERSION)
+	install -m 0755 $(SHAREDLIB) $(DESTDIR)$(LIBDIR)/$(SHAREDLIB).$(VERSION)
```

Comment:
On Fedora at least (I assume on all Linux distros) gcc produces .so files with mode 0755. And every .so file in my `/usr/lib64` directory is executable. I have to confess I don't know why exactly.

But it seems various scripts and programs don't treat the .so file as an actual shared library unless it is executable -- without this, the Fedora package does not work because it doesn't find debug symbols in the .so file nor does it find the soname as a virtual provides.

Reply:
ok, that is weird.